src/components/connect.js

```diff
@@ -88,7 +90,7 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
           this.finalMapStateToProps(state, props) :
           this.finalMapStateToProps(state)
 
-        return checkStateShape(stateProps)
+        return checkStateShape(this.constructor.displayName, stateProps)
```

Comment:
Should we use var before `wrapWithConnect`? Or move `checkStateShape` into `wrapWithConnect`? `checkStateShape` can't use `connectDisplayName` directly because they live in the different scope.